wgpu-core/src/device/resource.rs

```diff
@@ -854,7 +854,11 @@ impl Device {
                 user_closures.mappings,
                 user_closures.blas_compact_ready,
                 queue_empty,
-            ) = queue_result
+            ) = queue_result;
+            // Queue::drop is acquiring the snatch lock as well
+            drop(snatch_guard);
+        } else {
+            drop(snatch_guard);
```

Comment:
question: It seems like the potential for `Queue::drop` getting called here is when:

1. The user calls `Device::maintain` on this thread (let's call it thread 1) while holding a last strong ref. to the queue on another thread (let's call that thread 2).
2. Execution on _this_ thread (1) progresses until we have a strong ref. to `queue`, and follow the branch to call `queue.maintain(…)`.
3. Thread (2) drops its strong ref., making the strong ref. here in thread 1 the last one.

Does that sound right? Are you using `wgpu` from multiple threads?

Reply:
You're right, I tried to make the comment as explicit as possible but feel free to modify